eclipse-collections-code-generator/src/main/resources/impl/collection/mutable/abstractMultiReaderPrimitiveCollection.stg

```diff
@@ -0,0 +1,554 @@
+finally
+{
+    this.unlockWriteLock();
+}
+>>
+
+body(type, name) ::= <<
+<copyright()>
+
+package org.eclipse.collections.impl.collection.mutable.primitive;
+
+import java.io.Serializable;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.eclipse.collections.api.<name>Iterable;
+import org.eclipse.collections.api.Lazy<name>Iterable;
+import org.eclipse.collections.api.bag.primitive.Mutable<name>Bag;
+import org.eclipse.collections.api.block.function.primitive.<name>ToObjectFunction;
+import org.eclipse.collections.api.block.function.primitive.Object<name>ToObjectFunction;
+import org.eclipse.collections.api.block.predicate.primitive.<name>Predicate;
+import org.eclipse.collections.api.block.procedure.primitive.<name>Procedure;
+import org.eclipse.collections.api.collection.MutableCollection;
+import org.eclipse.collections.api.collection.primitive.Immutable<name>Collection;
+import org.eclipse.collections.api.collection.primitive.Mutable<name>Collection;
+import org.eclipse.collections.api.iterator.Mutable<name>Iterator;
+import org.eclipse.collections.api.iterator.<name>Iterator;
+import org.eclipse.collections.api.list.primitive.Mutable<name>List;
+import org.eclipse.collections.api.set.primitive.Mutable<name>Set;
+import org.eclipse.collections.impl.lazy.primitive.Lazy<name>IterableAdapter;
+import net.jcip.annotations.ThreadSafe;
+
+/**
+ * This file was automatically generated from template file abstractMultiReaderPrimitiveCollection.stg.
+ *
+ * @since 8.1.
+ */
+@ThreadSafe
+public abstract class AbstractMultiReader<name>Collection
+        implements Mutable<name>Collection, Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    private final ReadWriteLock lock;
+
+    private final Mutable<name>Collection collection;
+
+    protected AbstractMultiReader<name>Collection(Mutable<name>Collection collection)
+    {
+        this(collection, null);
+    }
+
+    protected AbstractMultiReader<name>Collection(Mutable<name>Collection collection, ReadWriteLock newLock)
+    {
+        if(collection == null)
+        {
+            throw new IllegalArgumentException("Cannot create a AbstractMultiReader<name>Collection on a null collection");
+        }
+        this.collection = collection;
+        this.lock = newLock == null ? new ReentrantReadWriteLock() : newLock;
+    }
+
+    public ReadWriteLock getLock()
+    {
+        return this.lock;
+    }
+
+
+    protected void acquireWriteLock()
+    {
+        this.getLock().writeLock().lock();
+    }
+
+    protected void unlockWriteLock()
+    {
+        this.getLock().writeLock().unlock();
+    }
+
+    protected void acquireReadLock()
+    {
+        this.getLock().readLock().lock();
+    }
+
+    protected void unlockReadLock()
+    {
+        this.getLock().readLock().unlock();
+    }
+
+
+    protected Mutable<name>Collection get<name>Collection()
+    {
+        return this.collection;
+    }
```

Comment:
If we follow the pattern that the object collections use, this method would be called getDelegate() and it would return PrimitiveIterable in the superclass.